common/src/tap/checks/timestamp_check.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+use anyhow::anyhow;
+use std::time::{Duration, SystemTime};
+
+pub struct TimestampCheck {
+    timestamp_error_tolerance: Duration,
+}
+
+use tap_core::receipt::{
+    checks::{Check, CheckResult},
+    Checking, ReceiptWithState,
+};
+
+impl TimestampCheck {
+    pub fn new(timestamp_error_tolerance: Duration) -> Self {
+        Self {
+            timestamp_error_tolerance,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Check for TimestampCheck {
+    async fn check(&self, receipt: &ReceiptWithState<Checking>) -> CheckResult {
+        let timestamp_now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)?;
+        let min_timestamp = timestamp_now - self.timestamp_error_tolerance;
+        let max_timestamp = timestamp_now + self.timestamp_error_tolerance;
+
+        let receipt_timestamp = Duration::from_nanos(receipt.signed_receipt().message.timestamp_ns);
+
+        if receipt_timestamp < max_timestamp && receipt_timestamp > min_timestamp {
+            Ok(())
+        } else {
+            Err(anyhow!(
+                "Receipt timestamp `{}` is outside of current system time +/- timestamp_error_tolerance",
+                receipt_timestamp.as_secs()
+            ))
+        }
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloy_primitives::Address;
+    use std::str::FromStr;
+
+    use tap_core::{
+        receipt::{checks::Check, Checking, Receipt, ReceiptWithState},
+        signed_message::EIP712SignedMessage,
+    };
+
+    use alloy_sol_types::eip712_domain;
+    use alloy_sol_types::Eip712Domain;
+    use ethers::signers::coins_bip39::English;
+    use ethers::signers::{LocalWallet, MnemonicBuilder};
+    fn create_signed_receipt_with_jitter(timestamp_ns: u64) -> ReceiptWithState<Checking> {
```

Comment:
You removed the built in jitter, so it needs renaming now
```suggestion
    fn create_signed_receipt(timestamp_ns: u64) -> ReceiptWithState<Checking> {
```